src/commands/deployment/onboard.test.ts

```diff
@@ -159,22 +170,122 @@ describe("validateRequiredArguments", () => {
 });
 
 describe("setConfiguration", () => {
-  test("Should pass when storage account and table names are specified", async () => {
-    const filename = path.resolve(mockFileName);
-    process.env.test_name = "testStorageName";
-    process.env.test_key = "testStorageKey";
+  test("Should pass updating previous storage account and table names", async () => {
+    try {
+      const data = {
+        introspection: {
+          azure: {
+            account_name: "test-storage",
+            table_name: "test-table"
+          }
+        }
+      };
+
+      // create config file in test location
+      fs.writeFileSync(testConfigFile, yaml.safeDump(data));
+
+      logger.info(`testConfigFile: ${testConfigFile}`);
+
+      // set storage and table names
+      await setConfiguration(storageAccountName, storageTableName);
+      loadConfiguration(testConfigFile);
+
+      const { azure } = Config().introspection!;
+      expect(azure!.account_name).toBe(storageAccountName);
+      expect(azure!.table_name).toBe(storageTableName);
+    } catch (err) {
+      logger.error(`dirs error: ${err}`);
+    }
+  });
+
+  test("Should pass updating previous undefined storage account and table names", async () => {
+    try {
+      const data = {
+        introspection: {
+          azure: {
+            account_name: undefined,
+            table_name: undefined
+          }
+        }
+      };
+
+      // create config file in test location
+      fs.writeFileSync(testConfigFile, yaml.safeDump(data));
```

Comment:
@sarath-p why do we need the config file?

Reply:
The test scenario is to check whether `onboard` can update an existing config file with storage account and table names are undefined/null and this particular line is storing the config file with empty values before updating.